edge/pkg/common/dbm/db_test.go

```diff
@@ -1,97 +1,116 @@
-package dbm_test
+package dbm
 
 import (
+	"errors"
+	"os"
 	"testing"
-
-	"github.com/kubeedge/beehive/pkg/common/log"
-	"github.com/kubeedge/kubeedge/edge/pkg/common/dbm"
-
-	"github.com/astaxie/beego/orm"
-	"github.com/stretchr/testify/assert"
 )
 
-const (
-	//MetaTable name
-	MetaTableName = "meta"
-	// DocumentTableName document table name
-	DocumentTableName = "document"
-)
-
-// Document device document
-type Document struct {
-	Deviceid   string `orm:"column(deviceid); size(64); pk"`
-	Expected   string `orm:"column(expected); type(text); null"`
-	Actual     string `orm:"column(actual); type(text); null"`
-	Metadata   string `orm:"column(metadata); type(text); null"`
-	Laststate  string `orm:"column(laststate); type(text); null"`
-	Lastsync   string `orm:"column(lastsync); type(text); null"`
-	Attributes string `orm:"column(attributes); type(text); null"`
-}
-
-// SaveDocument save document to db
-func SaveDocument(doc *Document) error {
-	_, err := dbm.DBAccess.Insert(doc)
-	return err
-}
-
-// Meta metadata object
-type TestMeta struct {
-	// ID    int64  `orm:"pk; auto; column(id)"`
-	Key   string `orm:"column(key); size(36); pk"`
-	Type  string `orm:"column(type); size(36)"`
-	Value string `orm:"column(value); null; type(text)"`
-}
-
-// SaveMeta save meta to db
-func SaveMeta(meta *TestMeta) error {
-	num, err := dbm.DBAccess.Insert(meta)
-	log.LOGGER.Debugf("Insert affected Num: %d, %s", num, err)
-	return err
+// TestDevice is a dummy struct that is used for model creation in orm.
+type TestDevice struct {
+	ID          string `orm:"column(id); size(64); pk"`
+	Name        string `orm:"column(name); null; type(text)"`
+	Description string `orm:"column(description); null; type(text)"`
+	State       string `orm:"column(state); null; type(text)"`
+	LastOnline  string `orm:"column(last_online); null; type(text)"`
 }
 
-// DeleteMetaByKey delete meta by key
-func DeleteMetaByKey(key string) error {
-	num, err := dbm.DBAccess.QueryTable(MetaTableName).Filter("key", key).Delete()
-	if err != nil {
-		log.LOGGER.Errorf("Something wrong when inserting data: %v", err)
-		return err
+// TestRegisterModel is function to test RegisterModel().
+func TestRegisterModel(t *testing.T) {
+	tests := []struct {
+		name       string
+		modulename string
+		m          interface{}
+	}{
+		{
+			//Failure Case
+			name:       "TestRegisterModel-UnregisteredModule",
+			modulename: "testmodule",
+			m:          "",
+		},
+		{
+			//Success Case
+			name:       "TestRegisterModel-RegisteredModule",
+			modulename: "twin",
+			m:          new(TestDevice),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			RegisterModel(test.modulename, test.m)
```

Comment:
There is no assert statement here, Doesn't make sense to call a function and not assert something in a testcase.

Reply:
The function gives an internal call to isModuleEnabled() function. Apart from this it registers the orm model. Here I have covered both scenarios of isModuleEnabled(). We can test this one separately as well. @sids-b , thoughts please.